libyul/AsmPrinter.cpp

```diff
@@ -228,3 +233,17 @@ string AsmPrinter::appendTypeName(YulString _type, bool _isBoolLiteral) const
 	else
 		return ":" + _type.str();
 }
+
+std::string AsmPrinter::formatSourceLocationComment(DebugData const& _debugData) const
+{
+	if (m_lastLocation == _debugData.location || !m_nameTosourceIndex)
+		return "";
+
+	auto resultIter = m_nameTosourceIndex->find(_debugData.location.sourceName);
+	if (resultIter == m_nameTosourceIndex->end())
+		return "";
+
+	m_lastLocation = _debugData.location;
+
+	return "/** @src " + to_string(resultIter->second) + ":" + to_string(_debugData.location.start) + ":" + to_string(_debugData.location.end) + " */\n";
```

Comment:
Maybe we should distinguish between statement context and expression context so that it formats nicer.